### PR TITLE
Graphattributes rnw

### PR DESCRIPTION
**Pull request by villafup**
@jwokaty for review.  

It was a little tricky resizing graph.  If it's too big I can try making it smaller.

[graphAttributes.pdf](https://github.com/Bioconductor/graph/files/10844864/graphAttributes.pdf)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -10,7 +10,7 @@ Authors@R: c(
         comment = "Converted 'MultiGraphClass' and 'GraphClass' vignettes from Sweave to RMarkdown / HTML."
     ),   
     person("Paul", "Villafuerte", role = "ctb",
-        comment = "Converted 'clusterGraph' vignettes from Sweave to RMarkdown / HTML."
+        comment = "Converted 'clusterGraph' and graphAttributes' vignettes from Sweave to RMarkdown / HTML."
```
Comment (sonali8434) [inline DESCRIPTION]: We can write here `comment = "Converted vignettes from Sweave to RMarkdown / HTML."` to keep it short if we convert more than one vignette in a package.

File: DESCRIPTION [hunk 2]
```diff
@@ -10,7 +10,7 @@ Authors@R: c(
...
     ),
     person("Bioconductor Package Maintainer",
         role = "cre",
```
Comment (sonali8434) [inline DESCRIPTION]: I think we need to fix the formatting little bit here you may write:

```
person("Paul", "Villafuerte", role = "ctb",
                comment = "Converted vignettes from Sweave to RMarkdown / HTML."
        ),
person("Bioconductor Package Maintainer", role = "cre",
                email = "maintainer@bioconductor.org"
        ))

File: vignettes/graphAttributes.Rmd [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+```
+
+```{r foo, fig.cap="The graph `g1`.", out.extra='id="foo"'}
+if (require("Rgraphviz")) {
+   gn = as(g1, "graphNEL")
+   plot(gn, nodeAttrs=makeNodeAttrs(gn, shape="circle", fillcolor="orange"))
+} else {
+ plot(1, 1, main="Rgraphviz required for this plot")
+}
+```
```
Comment (sonali8434) [inline vignettes/graphAttributes.Rmd]: I think this code block can be modified to get a few correction as we see there is no loading messages are there in original pdf also we see that this code block is hidden in original pdf so we use `echo=Flase` for this purpose also the height and width parameters are given in .Rnw file so I think we should include that too. 

the modified code chunk should look like 
```
```{r foo, fig.cap="The graph `g1`.", fig.height=6, fig.small=TRUE, fig.width=6, echo=FALSE, message=FALSE, out.extra='id="foo"'}
if (require("Rgraphviz")) {
   gn = as(g1, "graphNEL")
   plot(gn, nodeAttrs=makeNodeAttrs(gn, shape="circle", fillcolor="orange"))
} else {
 plot(1, 1, main="Rgraphviz required for this plot")
}
```
```

Comment (sonali8434) [reply vignettes/graphAttributes.Rmd]: I think we also need some help here @jwokaty the original pdf have these lines
```
A graphAM graph with undirected edges
Number of Nodes = 4
Number of Edges = 5
```
but it is not present in the new .html document how to solve this issue?

**Review by sonali8434 — CHANGES_REQUESTED** [review]
I am doing my first review @villafup I have left 4 reviews I don't think any more changes are required please do the requested modifications. After you made the requested changes I will do R CMD build in this branch to make it builds without any error and timeouts and also I which for tarball contains html.

Also I think we need merge conflicts.

**Comment by sonali8434** [comment]
We will this checklist to review your PR
- [x] the `.Rmd` file knits to `HTML`
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- In the `DESCRIPTION` file
  - [x] `BiocStyle` and `knitr` are listed in `Suggests`
  - [x] the line `VignetteBuilder: knitr` exists
  - [ ] any added lines use the same spacing and indents as the existing document
- If agreed with the Maintainer
  - [x] the contributor is in the author list in the `DESCRIPTION` file*.
  - [x] the contributor is in the author list in the vignette's YAML*.
- If this pull request involves converting from separate `Author` and  `Maintainer`
  lines to the `Authors@R` vector, please ensure that
  - [x] the `Authors@R` vector includes the maintainer as specified with `role='cre'`.
  - [x] the `Maintainer` line is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [x] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [x] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`
- [x] Only files necessary for the conversion are included in the pull request.
- [x] Long lines have been broken up and reformatted, you can achieve this by
  selecting the text in your script and then clicking on `Code > Reflow Comment`
  in R Studio.
- [x] Code blocks are surrounded by a blank line, and there are no blank lines
  before the start or after the end of the code within the code block. 

* Refer to the contribution guide for acceptable formats

File: DESCRIPTION (hunk 1 above)
Comment (villafup) [reply DESCRIPTION]: Done.  Also saw that the entry on line 10 also has 2 specific vignettes mentioned as well.

File: DESCRIPTION (hunk 2 above)
Comment (villafup) [reply DESCRIPTION]: Done.  is this this (and the one before) going to be the standard?  I think we'll have to go through all those we converted to make sure they follow the format.

**Review by sonali8434 — APPROVED** [review]

**Comment by sonali8434** [comment]
@jwokaty Please review Thanks @villafup for the changes.

File: vignettes/graphAttributes.Rmd (hunk 3 above)
Comment (jwokaty) [reply vignettes/graphAttributes.Rmd]: I had to ask how to do this. You should put parenthesis around the expression that you want returned. In this case, @villafup should do

    ```{r exampleGraph2, eval=TRUE}
    (g1 <- graphAM(adjMat=mat))
    ```

File: vignettes/graphAttributes.Rmd [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+mat <- matrix(c(0, 0, 1, 1, 
+                0, 0, 1, 1, 
+                1, 1, 0, 1, 
+                1, 1, 1, 0),
+              byrow=TRUE, ncol=4)
+rownames(mat) <- letters[1:4]
+colnames(mat) <- letters[1:4]
+```
+
+```{r exampleGraph2, print=TRUE}
```
Comment (jwokaty) [inline vignettes/graphAttributes.Rmd]: I don't think `print=TRUE` is a valid chunk option in R Markdown. Please remove it.

File: vignettes/graphAttributes.Rmd [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+nodeData(g1, n="a")
+nodeData(g1, n="a", attr="weight") <- 100
+nodeData(g1, n=c("a", "b"), attr="weight")
+nodeData(g1, n=c("a", "b"), attr="weight") <- 500
+nodeData(g1, n=c("a", "b"), attr="weight")
+nodeData(g1, n=c("a", "b"), attr="weight") <- c(11, 22)
+nodeData(g1, n=c("a", "b"), attr="weight")
+```
+
+```{r other, hide=TRUE, echo=FALSE}
```
Comment (jwokaty) [inline vignettes/graphAttributes.Rmd]: `hide=TRUE` isn't a valid chunk option. Let's remove `hide=TRUE, `.

**Review by jwokaty — CHANGES_REQUESTED** [review]
Very small changes just to remove chunk options that aren't valid in R Markdown then I think this will be ready to merge. Thanks for this PR.

**Review by jwokaty — APPROVED** [review]

**Pull request merged**